Cannot find `model_{RUN_ID}_latest.json` during inference on other HPC

### What happened?

After uploading a run from `santis` to S3 with
```
../WeatherGenerator-private/hpc/upload_experiment.py --run-id atmofs01
```
and downloading it to `Leonardo` with
```
../WeatherGenerator-private/hpc/download_experiment.py --run-id atmofs01
```
I'm trying to run inference on branch `mk/develop/forecast_release_v0` with
```
uv run --offline inference --from-run-id atmofs01 --samples 2 --options training_config.forecast.num_steps=40
```
but cannot find the `model_atmofs01_latest.json` and break with the following error message
```
[mkarlbau@lrdn1186 WeatherGenerator]$ uv run --offline inference --from-run-id atmofs01 --samples 2 --options training_config.forecast.num_steps=40
Traceback (most recent call last):
  File "/leonardo/home/userexternal/mkarlbau/repos/WeatherGenerator/.venv/bin/inference", line 10, in <module>
    sys.exit(inference())
             ^^^^^^^^^^^
  File "/leonardo/home/userexternal/mkarlbau/repos/WeatherGenerator/src/weathergen/run_train.py", line 42, in inference
    main([cli.Stage.inference] + sys.argv[1:])
  File "/leonardo/home/userexternal/mkarlbau/repos/WeatherGenerator/src/weathergen/run_train.py", line 59, in main
    run_inference(args)
  File "/leonardo/home/userexternal/mkarlbau/repos/WeatherGenerator/src/weathergen/run_train.py", line 99, in run_inference
    cf = config.load_merge_configs(
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/leonardo/home/userexternal/mkarlbau/repos/WeatherGenerator/packages/common/src/weathergen/common/config.py", line 394, in load_merge_configs
    base_config = load_run_config(from_run_id, mini_epoch, None)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/leonardo/home/userexternal/mkarlbau/repos/WeatherGenerator/packages/common/src/weathergen/common/config.py", line 238, in load_run_config
    assert fname.exists(), (
           ^^^^^^^^^^^^^^
AssertionError: ('The fallback path to the model does not exist. Please provide a `model_path`.', PosixPath('/leonardo_work/AIFAC_5C0_154/weathergen/shared_work/models/atmofs01/model_atmofs01_latest.json'))
```

In `models/atmofs01/`, I can find `model_atmofs01.json`, but not the `latest`. When I rename `model_atmofs01.json` to `model_atmofs01_latest.json`, the inference starts properly.

### What are the steps to reproduce the bug?

See above

### Hedgedoc link to logs and more information. This ticket is public, do not attach files directly.

_No response_

--- a/packages/common/src/weathergen/common/config.py
+++ b/packages/common/src/weathergen/common/config.py
@@ -234,12 +234,25 @@ def load_run_config(run_id: str, mini_epoch: int | None, model_path: str | None)
         else:
             path = Path(model_path) / run_id
 
-        fname = path / _get_model_config_file_read_name(run_id, mini_epoch)
-        assert fname.exists(), (
-            "The fallback path to the model does not exist. Please provide a `model_path`.",
-            fname,
-        )
-        _logger.info(f"Loading config from specified run_id and mini_epoch: {fname}")
+        config_path_with_epoch = path / _get_model_config_file_read_name(run_id, mini_epoch)
+        config_path_without_epoch = path / _get_model_config_file_read_name(run_id, None)
+
+        if config_path_with_epoch.exists():
+            fname = config_path_with_epoch
+            _logger.info(f"Loading config from specified run_id and mini_epoch: {fname}")
+        elif config_path_without_epoch.exists():
+            fname = config_path_without_epoch
+            _logger.info(
+                f"Config for mini_epoch {mini_epoch} not found. "
+                f"Falling back to config without mini_epoch: {fname}"
+            )
+        else:
+            raise FileNotFoundError(
+                f"Could not find model config for run_id '{run_id}' "
+                f"(mini_epoch={mini_epoch}) in '{path}'. "
+                f"Tried: '{config_path_with_epoch.name}' and '{config_path_without_epoch.name}'. "
+                f"Please provide a valid `model_path`."
+            )
 
     with fname.open() as f:
         json_str = f.read()

--- a/packages/common/src/weathergen/common/config.py
+++ b/packages/common/src/weathergen/common/config.py
@@ -251,7 +251,7 @@ def load_run_config(run_id: str, mini_epoch: int | None, model_path: str | None)
                 f"Could not find model config for run_id '{run_id}' "
                 f"(mini_epoch={mini_epoch}) in '{path}'. "
                 f"Tried: '{config_path_with_epoch.name}' and '{config_path_without_epoch.name}'. "
-                f"Please provide a valid `model_path`."
+                f"Please check run-id and mini_epoch."
             )
 
     with fname.open() as f:

--- a/packages/common/src/weathergen/common/config.py
+++ b/packages/common/src/weathergen/common/config.py
@@ -251,7 +251,7 @@ def load_run_config(run_id: str, mini_epoch: int | None, model_path: str | None)
                 f"Could not find model config for run_id '{run_id}' "
                 f"(mini_epoch={mini_epoch}) in '{path}'. "
                 f"Tried: '{config_path_with_epoch.name}' and '{config_path_without_epoch.name}'. "
-                f"Please check run-id and mini_epoch."
+                f"Please check run_id and mini_epoch."
             )
 
     with fname.open() as f:

